.werft/post-process.sh

```diff
@@ -0,0 +1,304 @@
+   SIZE="${#PORT}"
+   NAME=$(yq r k8s.yaml -d "$i" metadata.name)
+   KIND=$(yq r k8s.yaml -d "$i" kind)
+   if [[ "$SIZE" -ne "0" ]] && [[ "$NAME" == "registry-facade" ]] && [[ "$KIND" == "DaemonSet" ]] ; then
+      echo "setting $NAME to $REG_DAEMON_PORT"
+      yq w -i k8s.yaml -d "$i" spec.template.spec.containers.[0].ports.[0].hostPort "$REG_DAEMON_PORT"
+   fi
+   if [[ "$SIZE" -ne "0" ]] && [[ "$NAME" == "ws-daemon" ]] && [[ "$KIND" == "DaemonSet" ]] ; then
+      echo "setting $NAME to $WS_DAEMON_PORT"
+      yq w -i k8s.yaml -d "$i" spec.template.spec.containers.[0].ports.[0].hostPort "$WS_DAEMON_PORT"
+      yq w -i k8s.yaml -d "$i" spec.template.spec.containers.[0].ports.[0].containerPort "$WS_DAEMON_PORT"
+   fi
+
+   # override details for registry-facade service
+   if [[ "registry-facade" == "$NAME" ]] && [[ "$KIND" == "Service" ]]; then
+      WORK="overrides for $NAME $KIND"
+      echo "$WORK"
+      yq w -i k8s.yaml -d "$i" spec.ports[0].port "$REG_DAEMON_PORT"
+   fi
+
+   # override labels for pod scheduling on nodes
+   WORKSPACE_COMPONENTS=("image-builder" "image-builder-mk3 blobserve registry-facade" "ws-daemon" "agent-smith")
+   # shellcheck disable=SC2076
+   if [[ " ${WORKSPACE_COMPONENTS[*]} " =~ " ${NAME} " ]] && { [[ "$KIND" == "Deployment" ]] || [[ "$KIND" == "DaemonSet" ]]; }; then
+      LABEL="gitpod.io/workspace_$NODE_POOL_INDEX"
+      echo "setting $LABEL for $NAME"
+      touch /tmp/"$NAME"pool.yaml
+      # create a matching expression
+      yq w -i /tmp/"$NAME"pool.yaml spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key "$LABEL"
+      yq w -i /tmp/"$NAME"pool.yaml spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].operator Exists
+      # append it
+      yq m --arrays=overwrite -i k8s.yaml -d "$i" /tmp/"$NAME"pool.yaml
+   elif [[ "$KIND" == "DaemonSet" ]] || [[ "$KIND" == "Deployment" ]] || [[ "$KIND" == "StatefulSet" ]] || [[ "$KIND" == "Job" ]]; then
+      LABEL="gitpod.io/workload_meta"
+      echo "setting $LABEL for $NAME"
+      touch /tmp/"$NAME"pool.yaml
+      # create a matching expression
+      yq w -i /tmp/"$NAME"pool.yaml spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key "$LABEL"
+      yq w -i /tmp/"$NAME"pool.yaml spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].operator Exists
+      # append it
+      yq m --arrays=overwrite -i k8s.yaml -d "$i" /tmp/"$NAME"pool.yaml
+   fi
+
+   # overrides for server-config
+   if [[ "server-config" == "$NAME" ]] && [[ "$KIND" == "ConfigMap" ]]; then
+      WORK="overrides for $NAME $KIND"
+      echo "$WORK"
+      touch /tmp/"$NAME"overrides.yaml
```

Comment:
After reading in detail, this is less stuff than I feared. It's a bit noisy (not necessarily the bad kind) but most of it is connected to the devstaging setup.